core/electrum_client/src/types.rs

```diff
@@ -0,0 +1,201 @@
+use bitcoin::blockdata::block;
+use bitcoin::hashes::hex::FromHex;
+use bitcoin::hashes::{sha256, Hash};
+use bitcoin::{Script, Txid};
+
+use serde::{de, Deserialize, Serialize};
+
+static JSONRPC_2_0: &'static str = "2.0";
+
+#[derive(Serialize, Clone)]
+#[serde(untagged)]
+pub enum Param {
+    Usize(usize),
+    String(String),
+    Bool(bool),
+}
+
+#[derive(Serialize, Clone)]
+pub struct Request<'a> {
+    jsonrpc: &'static str,
```

Comment:
nit: I would call it `jsonrpc_version` or something like that

Reply:
it has to be called `jsonrpc` because that's its name in the [spec](https://www.jsonrpc.org/specification#request_object). i guess we could rename it in the struct and use serde to map it to a different name during serialization, but i don't think it's worth the effort